### ISSUE TITLE
texture loss is missing

### ISSUE THREAD

**Issue by ofir-dl** [widefield2sim/w2s#4]
Hi,

I have read your paper and saw that the texture loss is missing.

**According to the paper**: L = L 1 + α · L_perceptual + β · L_texture
**In the code**: loss = loss + 0.2*feature_factor*MSEcriterion(vgg_gt[i], vgg_output[i])

Do you plan to fix it?
In addition, what does the 0.2 factor stands for?

Thanks so much,
Ofir

**Comment by XuZikang** [comment]
Hi,

In my opinion, the _Ltexture has already included in the code. 

```python
for i in range(5):
    loss = loss + 0.2*feature_factor*MSEcriterion(vgg_gt[i], vgg_output[i])
    if (i == 4):
        loss += 0.2*feature_factor*MSEcriterion(CovpoolLayer(vgg_gt[i]), CovpoolLayer(vgg_output[i]))
```

The second line is perceptual loss and the fourth line represented texture loss (using `Convpoollayer`  declared in MPN-CONV )

In the paper, there is also some explanation:

> For the neural feature space, we use a pre-trained 19-layer VGG [39]. For the perceptual loss, we take the features from `conv1 2', `conv2 2' ,`conv3 4', `conv4 4' and `conv5 4'. For the texture loss, we take the features from `conv4 4'.

Also, I think the 0.2 factor is used to normalize these loss items, but I'm not sure.

Best Wishes,
Zikang Xu

**Comment by ofir-dl** [comment]
Ok, that makes sesne.
Thanks!
